### PR TITLE
deep assign if nested fragment

### PR DESCRIPTION
**Pull request by edvinerikson**
#534 fix attempt

I am not sure it fixes all cases for this. However some guidance would be helpful since I'm not familiar with the codebase

### CODE REVIEW

**Comment by edvinerikson** [comment]
Also I have no idea how to fix the TS errors

File: src/data/diffAgainstStore.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import isArray = require('lodash.isarray');
 import isNull = require('lodash.isnull');
 import isObject = require('lodash.isobject');
 import has = require('lodash.has');
-import assign = require('lodash.assign');
+const deepAssign = require('deep-assign');
```
Comment (helfer) [inline src/data/diffAgainstStore.ts]: Use `import` here.

File: test/readFromStore.ts [hunk 2]
```diff
@@ -157,6 +157,84 @@ describe('reading from the store', () => {
     });
   });
 
+  it.only('runs a nested fragment with multiple fragments', () => {
```
Comment (helfer) [inline test/readFromStore.ts]: with `it.only` you're running just this single test!

File: test/readFromStore.ts [hunk 3]
```diff
@@ -157,6 +157,84 @@ describe('reading from the store', () => {
...
+                numberField
+              }
+            }
+          }
+        }
+      `,
+      rootId: 'abcd',
+    });
+
+    // The result of the query shouldn't contain __data_id fields
```
Comment (helfer) [inline test/readFromStore.ts]: Why is this comment here?

**Comment by stubailo** [comment]
Merging here: https://github.com/apollostack/apollo-client/pull/669

**Pull request closed without merging**